addon-manager/impl/src/main/java/org/jboss/forge/addon/manager/impl/ui/DistributionUpdateCommand.java

```diff
@@ -0,0 +1,140 @@
+import org.jboss.forge.addon.resource.FileResource;
+import org.jboss.forge.addon.ui.annotation.Command;
+import org.jboss.forge.addon.ui.annotation.predicate.NonGUIEnabledPredicate;
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.furnace.addons.Addon;
+import org.jboss.forge.furnace.util.OperatingSystemUtils;
+import org.jboss.forge.furnace.versions.SingleVersion;
+import org.jboss.forge.furnace.versions.Version;
+
+public class DistributionUpdateCommand
+{
+   @Inject
+   Addon currentAddon;
+   @Inject
+   private DependencyResolver dependencyResolver;
+   
+   // to not duplicate the code
+   private File forgeHomeDir = OperatingSystemUtils.getForgeHomeDir();
+   private File updateDir = new File(forgeHomeDir,".update");
+   
+
+   @Command(value="Forge: Update Distribution",categories={ "Forge", "Manage" }, help="Update the whole forge installation", enabled=NonGUIEnabledPredicate.class)
+   public Result updateDistribution(UIExecutionContext context) throws Exception
+   {
+      String forgeHomeDirPath = OperatingSystemUtils.getForgeHomeDir().getAbsolutePath();
+      
+      if(updateDir.exists()) {
+         return Results.fail("There is already pending update in this forge distribution."
+                  + " Restart forge to finish the update or "
+                  + "run update-distribution-cancel command to cancel the updating process");
+      }
+      if (!context.getPrompt().promptBoolean(
+               "Are you sure you want to continue? This command will delete current directories: addons, bin, lib"))
+      {
+         return Results.fail("Updating distribution was aborted.");
+      }
+
+      List<Coordinate> distributionVersions = dependencyResolver.resolveVersions(DependencyQueryBuilder
+               .create(CoordinateBuilder.create()
+                        .setGroupId("org.jboss.forge")
+                        .setArtifactId("forge-distribution")));
+      Coordinate latestDistributionCoordinate;
+      int i = 0;
+      do
+      {
+         i++;
+         if (distributionVersions.size() <= i)
+         {
+            return Results.fail("Have not found any online Final version of the distribution.");
+         }
+         latestDistributionCoordinate = distributionVersions.get(distributionVersions.size() - i);
+      }
+      while (latestDistributionCoordinate.isSnapshot());
+      
+      Version latestVersion = new SingleVersion(latestDistributionCoordinate.getVersion());
+      Version installedVersion = currentAddon.getId().getVersion();
+      if (latestVersion.compareTo(installedVersion) <= 0)
+      {
+         return Results.fail("Forge is already in the most updated version.");
+      }
+
+      Dependency dependency = dependencyResolver.resolveArtifact(DependencyQueryBuilder.create(CoordinateBuilder
```

Comment:
Message: "This Forge installation will be updated to X.X.X.Final"
Prompt: "Is that ok?"